HopsanGenerator/templates/fmu_hopsan.cpp

```diff
@@ -138,45 +148,86 @@ void hopsan_get_message(hopsan_message_callback_t message_callback, void* userSt
 
 double hopsan_get_real(int vr)
 {
-    return (*dataPtrs[vr]);
+    if(vr < <<<nports>>>) {
+        return (*dataPtrs[vr]);
+    }
+    if(realParametersMap.count(vr)) {
+        hopsan::HString value;
+        spCoreComponentSystem->getParameterValue(realParametersMap[vr],value);
+        bool ok;
+        return value.toDouble(&ok);
+    }
+    return -1;
 }
 
 int hopsan_get_integer(int vr)
 {
-    //Write code here
+    if(intParametersMap.count(vr)) {
+        hopsan::HString value;
+        spCoreComponentSystem->getParameterValue(intParametersMap[vr],value);
+        bool ok;
+        return value.toLongInt(&ok);
+    }
     return -1;
 }
 
 int hopsan_get_boolean(int vr)
 {
-    //Write code here
+    if(boolParametersMap.count(vr)) {
+        hopsan::HString value;
+        spCoreComponentSystem->getParameterValue(boolParametersMap[vr],value);
+        bool ok;
+        return value.toBool(&ok);
+    }
     return false;
 }
 
 const char* hopsan_get_string(int vr)
 {
-    //Write code here
+    if(stringParametersMap.count(vr)) {
+        hopsan::HString value;
+        spCoreComponentSystem->getParameterValue(stringParametersMap[vr],value);
+        bool ok;
+        return value.c_str();
+    }
     return "";
 }
 
 void hopsan_set_real(int vr, double value)
 {
-    (*dataPtrs[vr]) = value;
+    if(vr < <<<nports>>>) {
+        (*dataPtrs[vr]) = value;
+    }
+    else if(realParametersMap.count(vr)) {
+        spCoreComponentSystem->setParameterValue(realParametersMap[vr], to_hstring(value));
+    }
 }
 
 void hopsan_set_integer(int vr, int value)
 {
-    //Write code here
+    if(intParametersMap.count(vr)) {
+        std::ostringstream ss;
+        ss << value;
+        spCoreComponentSystem->setParameterValue(intParametersMap[vr], hopsan::HString(ss.str().c_str()));
+    }
```

Comment:
Does not to_hstring work with int?

Reply:
It should work with int, but the funny thing is that under the hood, it just calls stringstream just like this code, with the addition of setting a precision

Reply:
It does work, so I replaced it.